### ISSUE TITLE
compile error with imgui 

### ISSUE THREAD

**Issue by Seekerzero** [koide3/iridescence#61]
I am on ubuntu 20.04, having this problem during compile.
`zwhy@zwhy:~/slam_project/iridescence/build$ make -j
[  1%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/imgui/imgui_tables.cpp.o
[  2%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/gl3w/gl3w.cpp.o
[  4%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/imgui/imgui_demo.cpp.o
[  5%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/imgui/imgui.cpp.o
[  6%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/imgui/imgui_draw.cpp.o
[  8%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/ImGuizmo/ImGuizmo.cpp.o
[  9%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/imgui/imgui_widgets.cpp.o
[ 10%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/ImGuizmo/ImCurveEdit.cpp.o
[ 12%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/ImGuizmo/ImGradient.cpp.o
[ 13%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/ImGuizmo/ImSequencer.cpp.o
[ 15%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/implot/implot.cpp.o
[ 17%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/imgui/backends/imgui_impl_glfw.cpp.o
[ 17%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/implot/implot_items.cpp.o
[ 19%] Building CXX object CMakeFiles/gl_imgui.dir/thirdparty/imgui/backends/imgui_impl_opengl3.cpp.o
/home/zwhy/slam_project/iridescence/thirdparty/imgui/backends/imgui_impl_glfw.cpp: In function ‘void ImGui_ImplGlfw_UpdateMonitors()’:
/home/zwhy/slam_project/iridescence/thirdparty/imgui/backends/imgui_impl_glfw.cpp:776:9: error: ‘glfwGetMonitorWorkarea’ was not declared in this scope; did you mean ‘glfwGetMonitorName’?
  776 |         glfwGetMonitorWorkarea(glfw_monitors[n], &x, &y, &w, &h);
      |         ^~~~~~~~~~~~~~~~~~~~~~
      |         glfwGetMonitorName
make[2]: *** [CMakeFiles/gl_imgui.dir/build.make:244: CMakeFiles/gl_imgui.dir/thirdparty/imgui/backends/imgui_impl_glfw.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:85: CMakeFiles/gl_imgui.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
`

**Comment by koide3** [comment]
Did you try this?

```bash
sudo apt-get install -y libglm-dev libglfw3-dev libpng-dev libjpeg-dev libeigen3-dev libboost-filesystem-dev libboost-program-options-dev
```

**Comment by Seekerzero** [comment]
I think so, I installed the dependency before the compile.

**Comment by Seekerzero** [comment]
I also check the newest version of imgui, the glfwGetMonitorWorkarea is already gone from imgui_impl_glfw.cpp. please also check this:
`https://fsunuc.physics.fsu.edu/git/gwm17/imgui/commit/aa8e09d7f1481641792e3bb6419acda896af62ae?style=unified&whitespace=`
 I would suppose that the version you copy is corresponding to a specific version of glfw3. 
 But seems like the ImGuizmo is depended on this version of imgui. which make it impossible to update it.

**Comment by Seekerzero** [comment]
ok I solved it by manually compile the glfw 3.3.8 version, which is not the default version of ubuntu 20.04.
